### PR TITLE
feat: add req update script

### PR DESCRIPTION
**Pull request by LauJohansson**
## What type of PR is this? (check all applicable)
- Feature

### Overview
This PR adds a powershell script that can be runned and automatically detects libraries and fix their version in the .cfg.
The install_requrements.txt is then the only file that need to be updated, if you need to add or alter a dependency.

### What is the current behavior?
In the moment, the dependency versions are not locked.

### What is the new behavior?
Running this script both updates and lock the dependencies

## Related Issues (Link to the open issue)
#142 

## Link

https://github.com/marketplace/actions/create-pull-request

### CODE REVIEW

File: .github/workflows/update-dependencies.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+jobs:
+  upgrade-requirements:
+    runs-on: windows-latest # Since we are running a PowerShell script
+
+    steps:
+    - uses: actions/checkout@v2 # Checks-out your repository
+    
+    - name: Set up Git
+      run: |
+        git config --global user.name 'github-actions[bot]'
```
Comment (LauJohansson) [inline .github/workflows/update-dependencies.yml:16]: Not sure about this workflow yet

File: README.md [hunk 2]
```diff
@@ -132,6 +133,24 @@ Releases to PyPI is an Github Action which needs to be manually triggered.
 [![Release](https://github.com/spetlr-org/spetlr/actions/workflows/release.yml/badge.svg)](https://github.com/spetlr-org/spetlr/actions/workflows/release.yml)
 [![PyPI spetlr badge](https://img.shields.io/pypi/v/spetlr)](https://pypi.org/project/spetlr/)
 
+
+# Requirements and dependencies
+The library has three txt-files at the root of the repo. These files defines three levels of requirements:
+- `requirements_install.txt` - this file contains the required libraries to be able to install spetlr.
+- `test_requirements.txt` - libraries required to run unit- and integration tests
```
Comment (tbg-delegate) [inline README.md:140]: This file naming convention `test_requirements.txt` conflicts with standard `pytest` discoverability, which looks for files with a preceding `test*`-pattern in their name. Hence running pytest in the root will yield an error unless specifically set to ignore, but I would much rather just change the filename.

File: README.md [hunk 3]
```diff
@@ -132,6 +133,24 @@ Releases to PyPI is an Github Action which needs to be manually triggered.
...
+- `dev_requirements.txt` - libraries required in the development process in order to contribute to the repo
+
+All libraries and their dependencies are added with a fixed version to the configuration file `setup.cfg`.
```
Comment (tbg-delegate) [inline README.md:143]: This appears to conflict with the above statement about `requirements_intall.txt`, or at least duplicate?

File: README.md [hunk 4]
```diff
@@ -132,6 +133,24 @@ Releases to PyPI is an Github Action which needs to be manually triggered.
...
+
+To __upgrade__ the the dependencies in the `setup.cfg` file do the following:
+
+1. Create a new branch
+2. Run `upgrade_requirements.ps1` in your terminal
+3. Commit the changes the script has made to the cfg file. If there are no changes everything is up to date.
```
Comment (tbg-delegate) [inline README.md:149]:
```suggestion
3. Commit the changes the script has made to the cfg file. If there are no changes, everything is up to date.
```

File: README.md [hunk 5]
```diff
@@ -132,6 +133,24 @@ Releases to PyPI is an Github Action which needs to be manually triggered.
...
+4. The PR runs all tests and ensure that the library is compliant with any updates
+
+Note that if it is desired to upgrade a dependency, but not to its newest version, it is possible to set the desired version in the `install_requirements.txt`, then this will be respected by the upgrade script.
```
Comment (tbg-delegate) [inline README.md:152]: Again, it is unclear whether the requirements in `setup.cfg` and `install_requirements.txt` should be, and if yes, are aligned.

File: install_requirements.txt [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+Deprecated
+pyodbc
+importlib-metadata
+azure-cosmos
+timezonefinder==6.0.2
+more_itertools
+pycountry
+databricks-sdk
+# Numpy does not support 3.8 if above 1.24
+numpy<=1.24;python_version<="3.8"
```
Comment (tbg-delegate) [inline install_requirements.txt:12]: If we drop 9.3 LTS support, we can probably remove these requirements altogether, by simply requiring Python > 3.8? Just food for thought.

File: upgrade_requirements.ps1 [hunk 7]
```diff
@@ -0,0 +1 @@
+spetlr-freeze-req install_requirements.txt --cfg
```
Comment (tbg-delegate) [inline upgrade_requirements.ps1:1]: What does this do? Would be nice with a docstring.

**Review by tbg-delegate — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (LauJohansson) [reply README.md:140]: I will rename

File: install_requirements.txt (hunk 6 above)
Comment (LauJohansson) [reply install_requirements.txt:12]: Good to know: The databricks-sdk has some restrictions on the request package: `requests>=2.28.1,<3`. I know that conflicted also with some other packages. Just for the ones who are going to update these dependencies in the future.

File: upgrade_requirements.ps1 (hunk 7 above)
Comment (LauJohansson) [reply upgrade_requirements.ps1]: added

File: upgrade_requirements.ps1
Comment (tbg-delegate) [inline upgrade_requirements.ps1:1]: Great work on the docstring.

File: install_requirements.txt (hunk 6 above)
Comment (tbg-delegate) [reply install_requirements.txt]: Noted. Resolving this comment.

File: README.md (hunk 5 above)
Comment (tbg-delegate) [reply README.md]: Resolving as my comment is outdated.

**Review by tbg-delegate — APPROVED** [review]

File: upgrade_requirements.ps1
Comment (LauJohansson) [reply upgrade_requirements.ps1:1]: Thanks!

**Review by mrmasterplan — APPROVED** [review]

**Pull request merged**